tsfc/driver.py

```diff
@@ -395,7 +399,11 @@ def compile_expression_dual_evaluation(expression, to_element, *,
                 point_set = PointSet(pts)
                 config = kernel_cfg.copy()
                 config.update(point_set=point_set)
+                # evaluate expression at the points specified by point_set
                 expr, = fem.compile_ufl(expression, **config, point_sum=False)
+                # the point set free indices should now be free indices of the
+                # complied expression
+                assert all(i in expr.free_indices for i in point_set.indices)
```

Comment:
Here too, I think `assert set(point_set.indices) <= set(expr.indices)` makes the assertion clearer.

Reply:
Oh cool, much neater and introduces me to python's sets too :) . Changed in d711e722d19ce86ae5fca8b60e01492bd55e6ae6